src/webserver/HttpBase.h

```diff
@@ -401,27 +400,23 @@ handle_request(
 
             if (!result.contains("error"))
                 result["status"] = "success";
-
-            responseStr = boost::json::serialize(response);
         }
 
-        auto warningFlag = false;
         boost::json::array warnings;
-        if (!dosGuard.add(ip, responseStr.size()))
-        {
-            warnings.emplace_back("Too many requests");
-            warningFlag = true;
-        }
+        warnings.emplace_back(
+            "This is a clio server. clio only serves validated data. If you "
+            "want to talk to rippled, include 'ledger_index':'current' in your "
+            "request");
         auto lastPublishAge = context->etl->lastPublishAgeSeconds();
         if (lastPublishAge >= 60)
-        {
             warnings.emplace_back("This server may be out of date");
-            warningFlag = true;
-        }
-        // reserialize only if a warning was appended.
-        if (warningFlag)
+        result["warnings"] = warnings;
+        responseStr = boost::json::serialize(response);
+        if (!dosGuard.add(ip, responseStr.size()))
         {
-            result["warning"] = warnings;
+            warnings.emplace_back("Too many requests");
+            result["warnings"] = warnings;
```

Comment:
Same here, will warnings only be added when there are "Too many requests"?

Reply:
Aren't they added on line 413?